### ISSUE TITLE
Do some sort of canonicalization on Page URLs

### ISSUE THREAD

**Issue by Mr0grog** [edgi-govdata-archiving/web-monitoring-db#124]
@jschell42 was trying to query some pages from our DB at:

```
https://api.monitoring.envirodatagov.org/api/v0/pages?url=https://www.globalchange.gov/*
```

…which returned no results. Turns out we have all those pages as `http` (not `https`), so either of these would have worked, but were not obvious:

```
https://api.monitoring.envirodatagov.org/api/v0/pages?url=http://www.globalchange.gov/*

https://api.monitoring.envirodatagov.org/api/v0/pages?url=*://www.globalchange.gov/*
```

It might be less accurate (a domain *could* serve entirely different content on HTTP vs. HTTPS), but I wonder if we should either strip the protocol from the stored URL or always make the query param agnostic to protocol.

Internet Archive’s CDX API handles this by searching/matching on a SURT-formatted URL (a canonicalized version of the URL that does not include protocol, reverses domain order for easier sorting, and strips `www*` subdomains). The downside is that, even if you give it an exact URL to match on, you may get multiple results (since multiple exact URLs could map to the SURT URL), but the record you are looking for is more likely to be present.

**Comment by janakrajchadha** [comment]
I don't think that stripping the protocol from the stored URL is a good idea. If a domain could serve different content on different protocols, it's important that we save the protocol information as well.

If there aren't too many results from the CDX API matching in most cases, then I think it would be the best solution in this situation.

**Comment by janakrajchadha** [comment]
Some information about SURT - http://crawler.archive.org/articles/user_manual/glossary.html#surt
Might prove to be helpful in this discussion.

**Comment by Mr0grog** [comment]
Anecdotally, if you do a search like this on archive.org’s CDX API, you get a lot of different URLs when you think you are looking for an exact match, which is my main concern here. That said, we *currently* have a much cleaner database than archive.org could ever hope to in this regard. Don’t know if that situation will persist, though, especially as we start to pull in archive.org data.

**Comment by janakrajchadha** [comment]
Maybe we can go through the list of URLs returned by the API and look for an exact match of the record. We can then add a flag (something like `exact_match`) in the response and set it to True if we find an exact match and return that link alone. If we don't find an exact match, we can send back the entire list and set the flag to False. I'm not sure how searching through the list will affect the response time of our API. These are my thoughts after understanding the basic idea behind SURT formatted URLs. I'll probably be able to give a more pragmatic suggestion once I've read a bit more about it.

**Comment by danielballan** [comment]
Thanks for the link, @janakrajchadha. SURT is new jargon to me.

Stuff like this...

> Conversion to SURT form also involves making all characters lowercase, and changing the 'https' scheme to 'http'. 

...seems good for IA's use case ("I am a human who needs to find a copy of this archived thing...") but less good for ours, where exact routes may matter for making exact comparisons.

How about a new query parameter, `surt=`?

**Comment by Mr0grog** [comment]
I’m going to close this and consider it fixed as part of #237.
